test/test_url_validity.py

```diff
@@ -3,7 +3,7 @@
 from __future__ import print_function
 
 
-from io import BytesIO
+from io import StringIO
```

Comment:
Since the unit tests run with Python 2 this should not use the `io` package but `cStringIO` instead (http://wiki.ros.org/python_2_and_3_compatible_code#StringIO).